packages/worker/src/interceptors.ts

```diff
@@ -31,19 +31,55 @@ export interface ActivityInboundCallsInterceptor {
 
 /**
  * A function that takes Activity Context and returns an interceptor
+ *
+ * @deprecated Use {@link ActivityInterceptorsFactory} instead
  */
 export interface ActivityInboundCallsInterceptorFactory {
   (ctx: ActivityContext): ActivityInboundCallsInterceptor;
 }
 
+/** Input for ActivityOutboundCallsInterceptor.getLogAttributes */
+export type GetLogAttributesInput = Record<string, unknown>;
+
+/**
+ * Implement any of these methods to intercept Activity outbound calls
+ */
+export interface ActivityOutboundCallsInterceptor {
+  /**
+   * Called on each invocation of the `activity.log` methods.
+   *
+   * The attributes returned in this call are attached to every log message.
+   */
+  getLogAttributes?: (input: GetLogAttributesInput, next: Next<this, 'getLogAttributes'>) => Record<string, unknown>;
+}
+
+export interface ActivityInterceptors {
+  inbound?: ActivityInboundCallsInterceptor[];
+  outbound?: ActivityOutboundCallsInterceptor[];
+}
+
+/**
+ * A function that takes Activity Context and returns an interceptor
+ */
+export type ActivityInterceptorsFactory = (ctx: ActivityContext) => ActivityInterceptors;
+
 /**
  * Structure for passing in Worker interceptors via {@link WorkerOptions}
  */
 export interface WorkerInterceptors {
   /**
-   * List of factory functions returning {@link ActivityInboundCallsInterceptor}s
+   * List of factory functions that instanciate {@link ActivityInboundCallsInterceptor}s and
+   * {@link ActivityOutboundCallsInterceptor}s.
    */
-  activityInbound?: ActivityInboundCallsInterceptorFactory[];
+  activity?: ActivityInterceptorsFactory[];
```

Comment:
I think just a single factory is fine since each factory already returns an array of interceptors but I can see that this is consistent with workflows so I'm okay leaving it for the extra flexibility.

Reply:
Again, I think the opposite would be easier to work with, ie. accept an array of factory, but have each factory return at most one inbound and one outbound, but I went for consistency.

Reply:
Fixed.